contracts/0.8.25/vaults/StakingVault.sol

```diff
@@ -331,9 +348,27 @@ contract StakingVault is IStakingVault, OwnableUpgradeable {
                 deposit.signature,
                 deposit.depositDataRoot
             );
+
             totalAmount += deposit.amount;
+            concatenatedDepositDataRoots = abi.encodePacked(concatenatedDepositDataRoots, deposit.depositDataRoot);
         }
 
+        if (
+            !SignatureUtils.isValidSignature(
```

Comment:
Why does signature verification occur after sending a deposit to the beacon chain? ```BEACON_CHAIN_DEPOSIT_CONTRACT.deposit{value: deposit.amount}(...args)```
Shouldn't it be the other way around?